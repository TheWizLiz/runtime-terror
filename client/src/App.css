.App {
    text-align: center;
}

.dropdown {
    position: relative;
}