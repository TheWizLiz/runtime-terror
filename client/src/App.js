import React from "react";
import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
import { Navigation, Footer, Home, About, Games, PlayerSafetyPlan, PvPRules, CashHuntRules, ModGuide, Updates, Leaderboard, Contact, LogIn, Forgot, Registration, Recover, AccountDetails, AdminDashboard, GameCreation, ActionLog, PropertyLog, SignUp} from "./components";
function App() {
  return (
    <div className="App">
      <Router>
        <Navigation />
        <Switch>
          <Route path="/" exact component={() => <Home />} />
          <Route path="/about" exact component={() => <About />} />
          <Route path="/games" exact component={() => <Games />} />
          <Route path="/mod-guide" exact component={() => <ModGuide />} />
          <Route path="/cash-hunt-rules" exact component={() => <CashHuntRules />} />
          <Route path="/player-safety-plan" exact component={() => <PlayerSafetyPlan />} />
          <Route path="/pvp-rules" exact component={() => <PvPRules />} />
          <Route path="/updates" exact component={() => <Updates />} />
          <Route path="/leaderboard" exact component={() => <Leaderboard />} />
          <Route path="/contact" exact component={() => <Contact />} />
          <Route path="/login" exact component={() => <LogIn />} />
          <Route path="/recover/:user" exact component={() => <Recover />} />
          <Route path="/registration/:id" exact component={() => <Registration />} />
          <Route path="/forgot" exact component={() => <Forgot />} />
          <Route path="/account-details" exact component={() => <AccountDetails />} />
          <Route path="/admin-dashboard" exact component={() => <AdminDashboard />} />
          <Route path="/game-creation" exact component={() => <GameCreation />} />
          <Route path="/action-log" exact component={() => <ActionLog />} />
          <Route path="/property" exact component={() => <PropertyLog />} />
          <Route path="/signup" exact component={() => <SignUp />} />
        </Switch>
        <Footer />
      </Router>
    </div>
  );
}

export default App;