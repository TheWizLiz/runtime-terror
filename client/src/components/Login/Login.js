import React from 'react'
import LoginForm from './LoginForm'
import SignUpForm from './SignUpForm'

class Home extends React.Component {
  constructor (props) {
    super(props)

    this.state = {
      token: '',
      isLoading: true
    }
  }

  render () {
    if (!this.state.token) {
      return (
        <div>
          <p>Sign Up</p>
          <SignUpForm />
          <p>Sign In</p>
          <LoginForm />
        </div>
      )
    }
    // Should re-direct to Home Screen or Account Details when returned Login token
    // TODO: Add Logout Button.
    return (
      <div>
        <p>Account (Verified)</p>
      </div>
    )
  }
}

export default Home
