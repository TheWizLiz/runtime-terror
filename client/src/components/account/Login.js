import React from 'react'
import { getFromStorage } from '../utils/storage'
import { BrowserRouter as Router, Link } from "react-router-dom";
import LoginForm from './LoginForm'
import LogoutButton from './LogoutButton'

class Login extends React.Component {
  constructor (props) {
    super(props)

    this.state = {
      token: '',
      isLoading: false
    }
    this.loggedHandler = this.loggedHandler.bind(this)
  }

  loggedHandler () {
    this.setState({
      isLoading: true
    })
  }

  componentDidMount () {
    // Check if there is a token with a user session already
    // If true, then replace NavBar 'Login' with 'Logout'
    // Also, add AccountDetails to pop up on the navbar
    // Replace screen with show account details (verified and logout button)

    // If user has logged in before, 'the_main_app' will have a token
    // Otherwise, there will be nothing and it will return null.
    console.log('MOUNT')
    const storage = getFromStorage('the_main_app')
    if (storage && storage.token) {
      // If app already has a token, then user is already logged in.
      const { token } = storage
      this.setState({ token: token, isLoading: false })
    }
  }

  // Situations:
  // 1. User Logs in for the first time
  // a. isLoggedIn False, No Token => Token and LoggedIn
  // 2. User is already logged in
  // b. Token is there, no isLogged => token: token, isLogged: true; updates
  // 3. User is not logged in already
  // c. token is false, isLogged is false
  componentDidUpdate () {
    // If user logs in, isLoggedIn is true, token is true
    console.log('UPDATE')
    const storage = getFromStorage('the_main_app')
    if (storage && storage.token && this.state.isLoading) {
      const { token } = storage
      this.setState({ token: token, isLoading: false })
    } else if (!storage && this.state.token && this.state.isLoading) {
      // Occur when logoutButton
      this.setState({ token: '', isLoading: false })
    }
  }

  render () {
    if (!this.state.token) {
      return (
        <div>
          <p>Gator Humans vs Zombies</p>
          <LoginForm handler={this.loggedHandler} /> <br />
          <p>
            Don't have an account?
            <Link to='/signup'> Sign Up</Link> <br />
          </p>
          <Link to='/forgot'>Forgot Password?</Link>
        </div>
      )
    } else {
      return (
        <div>
          <p>Account (Verified)</p>
          <Link to='/account-details'>Account Details</Link> <br /> <br />
          <LogoutButton handler={this.loggedHandler} />
        </div>
      )
    }
    // Should re-direct to Home Screen or Account Details when returned Login token
    // TODO: Add Logout Button.
  }
}

export default Login
