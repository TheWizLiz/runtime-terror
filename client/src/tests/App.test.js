import { render, screen } from '@testing-library/react';
import TestRenderer from 'react-test-renderer'
import App from '../App.js';
import Login from '../components/Login/Login.js'

// NOT A VERY GOOD TEST...
test('Checking for Login Screen Rendering...', () => {
  const component = TestRenderer.create(<Login />).toJSON()
  expect(component).toMatchSnapshot()
})

test('Renders Home Text', () => {
  render(<App />);
  const linkElement = screen.getByText(/Home/i);
  expect(linkElement).toBeInTheDocument();
});
