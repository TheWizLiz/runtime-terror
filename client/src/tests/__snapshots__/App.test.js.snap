// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checking for Login Screen 1`] = `
<div>
  <p>
    Sign Up
  </p>
  <div
    className="signUpForm"
  >
    <p>
      
    </p>
    <form
      onSubmit={[Function]}
    >
      <div
        className="textForm"
      >
        <label>
          First Name:
          <input
            name="firstname"
            onChange={[Function]}
            placeholder="First Name"
            type="text"
            value=""
          />
        </label>
        <br />
        <label>
          Last Name:
          <input
            name="lastname"
            onChange={[Function]}
            placeholder="Last Name"
            type="text"
            value=""
          />
        </label>
        <br />
        <label>
          Email:
          <input
            name="email"
            onChange={[Function]}
            placeholder="Email"
            type="email"
            value=""
          />
        </label>
        <br />
        <label>
          Username:
          <input
            name="username"
            onChange={[Function]}
            placeholder="Username"
            type="text"
            value=""
          />
        </label>
        <br />
        <label>
          Password:
          <input
            name="password"
            onChange={[Function]}
            placeholder="Password"
            type="password"
            value=""
          />
        </label>
        <br />
        <input
          type="submit"
          value="Submit"
        />
      </div>
    </form>
  </div>
  <p>
    Sign In
  </p>
  <div
    className="signInForm"
  >
    <p>
      
    </p>
    <form
      onSubmit={[Function]}
    >
      <div
        className="loginForm"
      >
        <label>
          Email:
          <input
            name="email"
            onChange={[Function]}
            placeholder="Email"
            type="text"
            value=""
          />
        </label>
        <br />
        <label>
          Password:
          <input
            name="password"
            onChange={[Function]}
            placeholder="Password"
            type="password"
            value=""
          />
        </label>
        <br />
        <input
          type="submit"
          value="Submit"
        />
      </div>
    </form>
    <button
      onClick={[Function]}
    >
      Logout
    </button>
  </div>
</div>
`;
